src/surveySimPP/readers/CSVReader.py

```diff
@@ -0,0 +1,191 @@
+        """
+        pplogger = logging.getLogger(__name__)
+
+        # Skip the rows before the header and then begin_loc rows after the header.
+        skip_rows = []
+        if self.header_row > 0:
+            skip_rows = [i for i in range(0, self.header_row)]
+        if block_start > 0:
+            skip_rows.extend([i for i in range(self.header_row + 1, self.header_row + 1 + block_start)])
+
+        # Read the rows.
+        if self.sep == "whitespace":
+            res_df = pd.read_csv(
+                self.filename,
+                delim_whitespace=True,
+                skiprows=skip_rows,
+                nrows=block_size,
+            )
+        else:
+            res_df = pd.read_csv(
+                self.filename,
+                delimiter=",",
+                skiprows=skip_rows,
+                nrows=block_size,
+            )
+
+        # Strip out the whitespace from the column names.
+        res_df = res_df.rename(columns=lambda x: x.strip())
+
+        # Check that the ObjID column exists and convert it to a string.
+        try:
+            res_df["ObjID"] = res_df["ObjID"].astype(str)
+        except KeyError:
+            err_str = f"ERROR: Unable to find ObjID column headings in {self.filename}."
+            pplogger.error(err_str)
+            sys.exit(err_str)
+
+        # Check for NaNs or nulls.
+        if "validate_data" in kwargs and kwargs["validate_data"]:  # pragma: no cover
+            if res_df.isnull().values.any():
+                pdt = res_df[res_df.isna().any(axis=1)]
+                inds = str(pdt["ObjID"].values)
+                outstr = (
+                    f"ERROR: While reading {self.filename} found uninitialised values ObjID: {str(inds)}."
+                )
+                pplogger.error(outstr)
+                sys.exit(outstr)
+
+        return res_df
+
+    def _build_id_map(self):
+        """Builds a table of just the object IDs"""
+        if self.obj_id_table is not None:
+            return
+
+        if self.sep == "whitespace":
+            self.obj_id_table = pd.read_csv(
+                self.filename,
+                delim_whitespace=True,
+                usecols=["ObjID"],
+                header=self.header_row,
+            )
+        else:
+            self.obj_id_table = pd.read_csv(
+                self.filename,
+                delimiter=",",
+                usecols=["ObjID"],
+                header=self.header_row,
+            )
+
+    def read_objects(self, obj_ids, **kwargs):
+        """Read in a chunk of data for given object IDs.
+
+        Parameters:
+        -----------
+        obj_ids (list): A list of object IDs to use.
+
+        Returns:
+        -----------
+        res_df (Pandas dataframe): The dataframe for the ephemerides.
+        """
+        self._build_id_map()
+
+        # Create list of only the matching rows for these object IDs and the header row.
+        row_good = [False] * self.header_row  # pre-header
```

Comment:
This is a minor point, feel free to ignore - given the way that the `row_good` list is used in the lambda, could we rename it? Something like the following?

```
skipped_row = [True] * self.header_row  # skip the pre-header
skipped_row.extend([False])  # Keep the the column header
skipped_row.extend(~self.obj_id_table["ObjId"].isin(obj_ids).values)

...

res_df = pd.read_csv( ...
    skiprows=(lambda x: skipped_row[x]),
)
```